homeassistant/components/sensor/recorder.py

```diff
@@ -280,3 +280,36 @@ def compile_statistics(
         result[entity_id]["stat"] = stat
 
     return result
+
+
+def list_statistic_ids(hass: HomeAssistant, statistic_type: str | None = None) -> dict:
+    """Return statistic_ids and meta data."""
+    entities = _get_entities(hass)
+
+    statistic_ids = {}
+
+    for entity_id, device_class in entities:
+        provided_statistics = DEVICE_CLASS_STATISTICS[device_class]
+
+        if statistic_type is not None and statistic_type not in provided_statistics:
+            continue
+
+        state = hass.states.get(entity_id)
+        assert state
+
+        if "sum" in provided_statistics and ATTR_LAST_RESET not in state.attributes:
+            continue
+
+        native_unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+
+        if device_class not in UNIT_CONVERSIONS:
+            statistic_ids[entity_id] = native_unit
```

Comment:
let's make this a dictionary for the value so it's possible to expand in the future.
```suggestion
            statistic_ids[entity_id] = {"unit": native_unit}
```